### PR TITLE
added tasks for provisioning/deprovisioning rosa hcp

### PR DESCRIPTION
**Pull request by xinredhat**

### CODE REVIEW

File: common/tasks/rosa-hcp/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+You need to create a secret named `hcp-cluster-secre` in Konflux, which includes the following data.
+
+* AWS_ACCOUNT_ID
+* AWS_OIDC_CONFIG_ID
+* INSTALL_ROLE_ARN
+* OPERATOR_ROLES_PREFIX
+* SUBNET_IDS
+* SUPPORT_ROLE_ARN
+* WORKER_ROLE_ARN
+* aws_access_key_id
```
Comment (flacatus) [inline common/tasks/rosa-hcp/README.md:20]: can we use uppercase for all environments?

File: common/tasks/rosa-hcp/generate-cluster-name.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  steps:
+    - name: generate-cluster-name
+      image: registry.access.redhat.com/ubi9/openssl:9.1-2
+      script: |
+        #!/usr/bin/env bash
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+        
+        CLUSTER_NAME="ci-$( openssl rand -hex 5 )"
```
Comment (flacatus) [inline common/tasks/rosa-hcp/generate-cluster-name.yaml:19]: CLUSTER_NAME="ci-$( openssl rand -hex 5 )"

File: common/tasks/rosa-hcp/hcp-cluster-delete.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: hcp-cluster-delete-task
+spec:
+  params:
+    - name: name
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-delete.yaml:8]: cluster-name will be more descriptive

File: common/tasks/rosa-hcp/hcp-cluster-create.yaml [hunk 4]
```diff
@@ -0,0 +1,207 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: hcp-cluster-create-task
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-create.yaml:5]: rosa-hcp-provision

File: common/tasks/rosa-hcp/hcp-cluster-create.yaml [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+spec:
+  results:
+    - name: ocp-login-command
+      description: The command to login to the created cluster
+  params:
+    - name: version
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-create.yaml:11]: will be better to rename to ocp-version

File: common/tasks/rosa-hcp/hcp-cluster-delete.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      description: ocp cluster name that you want to delete
+    - name: region
+      description: ocp cluster region where you want to delete
+  steps:
+    - name: delete-cluster
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-delete.yaml:13]: deprovision

File: common/tasks/rosa-hcp/generate-cluster-name.yaml (hunk 2 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/generate-cluster-name.yaml:19]: @flacatus what do you mean? it's totally same as mine.

File: common/tasks/rosa-hcp/hcp-cluster-create.yaml [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+                exit 1
+            fi
+        }
+
+        deploy_cluster() {
+            printf "INFO: Log in to your Red Hat account...\n" 
+            config_aws_creds
+            rosa login --token="$ROSA_TOKEN"
+
+            printf "Create ROSA with HCP cluster...\n"
```
[the remark this answers is not in the document]
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-create.yaml]: fixed

File: common/tasks/rosa-hcp/hcp-cluster-delete.yaml (hunk 3 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-delete.yaml]: fixed

File: common/tasks/rosa-hcp/hcp-cluster-create.yaml (hunk 4 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-create.yaml]: fixed

File: common/tasks/rosa-hcp/hcp-cluster-delete.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: hcp-cluster-delete-task
```
[the remark this answers is not in the document]
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-delete.yaml]: fixed

File: common/tasks/rosa-hcp/hcp-cluster-delete.yaml (hunk 6 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-delete.yaml]: fixed

File: common/tasks/rosa-hcp/hcp-cluster-create.yaml (hunk 5 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-create.yaml]: fixed

File: common/tasks/rosa-hcp/README.md (hunk 1 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/README.md:20]: fixed

File: common/tasks/rosa-hcp/generate-cluster-name.yaml (hunk 2 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/generate-cluster-name.yaml]: fixed.

File: common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        steps:
+          - name: run-tests
+            image: quay.io/konflux-qe-incubator/konflux-qe-tools:latest
+            script: |
+              set -o errexit
+              set -o nounset
+              set -o pipefail
+              echo "Running tests..."
+              echo "$(params.ocp-login-command)"
+    - name: delete-rosa-cluster
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml:64]: Lets use finally, if a previous task failed, deprovision will never happen

File: common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+      runAfter:
+        - test
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/xinredhat/konflux-qe-definitions.git
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml:71]: Dont use your fork values

File: common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    - name: ocp-version
+    - name: region
+    - name: hcp-secrets
+  tasks:
+    - name: generate-cluster-name
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/xinredhat/konflux-qe-definitions.git
```
Comment (flacatus) [inline common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml:17]: Dont use fork values

File: common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml (hunk 10 above)
Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml:71]: good catch, I forgot to update it after verfication

Comment (xinredhat) [reply common/tasks/rosa-hcp/hcp-cluster-test-pipeline.yaml]: fixed.

**Comment by flacatus** [comment]
@xinredhat let's move the pipeline definition to `integration-test/rhtap/rhtap-installer/pipelines`

**Review by flacatus — APPROVED** [review]

File: common/tasks/rosa-hcp/generate-cluster-name.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  steps:
+    - name: generate-cluster-name
+      image: registry.access.redhat.com/ubi9/openssl:9.1-2
+      script: |
+        #!/usr/bin/env bash
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+        
+        CLUSTER_NAME="kx-$( openssl rand -hex 5 )"
```
Comment (ascerra) [inline common/tasks/rosa-hcp/generate-cluster-name.yaml:19]: @xinredhat  What do you think about using the pipelinerun name to get a string for the cluster name. Then we will be able to easily map which cluster comes from which run.

Comment (flacatus) [reply common/tasks/rosa-hcp/generate-cluster-name.yaml:19]: @ascerra pipelinerun name unfortunately is to long for rosa naming. I think needs 15 chars and pr run have more

Comment (ascerra) [reply common/tasks/rosa-hcp/generate-cluster-name.yaml:19]: Right I was thinking we could just cut the output to use only the last 5 characters

Comment (flacatus) [reply common/tasks/rosa-hcp/generate-cluster-name.yaml:19]: makes sense for me. Feel free to open a pr

**Pull request merged**